src/main/java/pl/wedding/management/service/user/InMemoryRepository.java

```diff
@@ -0,0 +1,76 @@
+package pl.wedding.management.service.user;
+
+import pl.wedding.management.service.user.model.User;
+
+import java.util.*;
+import java.util.function.Function;
+
+final class InMemoryRepository implements UserRepository {
+
+    private static final Set<User> MEMORY_USER_REPOSITORY = new HashSet<>();
+
+    @Override
+    public Optional<User> findUserById(Integer id) {
+
+        return findByObjectValue(id, User::getUserId);
+    }
+
+    @Override
+    public Optional<User> findUserByEmail(String email) {
+
+        return findByObjectValue(email, User::getEmail);
+    }
+
+    @Override
+    public Integer addUser(User user) {
+
+        if(MEMORY_USER_REPOSITORY.contains(user)){
+            throw new IllegalStateException("User already exists!");
+        }
+
+        User userWithId = User.builder()
+                .userId(getLastIndexOfRepository())
+                .email(user.getEmail())
+                .password(user.getPassword())
+                .build();
+        MEMORY_USER_REPOSITORY.add(userWithId);
+
+        return userWithId.getUserId();
+    }
+
+    @Override
+    public boolean removeUserById(Integer id) {
+
+        Optional<User> user = findByObjectValue(id, User::getUserId);
+        return user.map(MEMORY_USER_REPOSITORY::remove)
+                .orElse(false);
+    }
+
+    @Override
+    public Collection<User> getAllUsers() {
+        return MEMORY_USER_REPOSITORY;
```

Comment:
you are returning reference to your repository here. Anyone with that reference can modify your repository. I think the best solution here would be return repository copy
```suggestion
        return new HashSet(MEMORY_USER_REPOSITORY);
```